### PR TITLE
DRAFT: 54 dataset

### PR DESCRIPTION
**Pull request by giorgiapitteri**
PR for implementing a dataset with the right label associated to the image.

Previously the get_item method returned only the MNISt image with a dummy label always set to 0. Now the correct label is return.

### CODE REVIEW

File: src/explib/datasets.py [hunk 1]
```diff
@@ -21,7 +21,9 @@ class DequantizedDataset(torch.utils.data.Dataset):
     def __init__(
         self,
         dataset: T.Union[os.PathLike, torch.utils.data.Dataset, np.ndarray],
```
Comment (giorgiapitteri) [inline src/explib/datasets.py:23]: Should we also specify here that dataset can be a torch.Tensor as well?

File: src/explib/datasets.py [hunk 2]
```diff
@@ -30,6 +32,13 @@ def __init__(
         else:
             self.dataset = pd.read_csv(dataset).values
 
+        #
+        self.dataset = self.dataset.to(device)
+        if not isinstance(labels, torch.Tensor):
```
Comment (giorgiapitteri) [inline src/explib/datasets.py:37]: If dataset is a torch.utils.data.Dataset we don't need labels right? It automatically has a getitem method that returns the data and the label. So I would need to add a check here.

Comment (giorgiapitteri) [reply src/explib/datasets.py:37]: I think The baseclass __getitem__ method is going to fail if the dataset is only a np array or a torch.Tensor. I understand there is the correct implementation in the derived class but maybe here we would need to put a check or raise an error?

File: src/explib/datasets.py [hunk 3]
```diff
@@ -21,8 +21,9 @@ class DequantizedDataset(torch.utils.data.Dataset):
     def __init__(
         self,
         dataset: T.Union[os.PathLike, torch.utils.data.Dataset, np.ndarray],
```
Comment (giorgiapitteri) [inline src/explib/datasets.py:23]: Would it be better to have a dataset object like a dictionary with images/labels? Otherwise I would renamed this "images" instead of dataset since we now have the labels object as well.
